### PR TITLE
docs: update component style conventions doc

### LINKED ISSUE #118: Docs: Update Component Theming guide
Type: **minor**

**Describe the bug**
Update [component theming guide](https://github.com/reactioncommerce/reaction-component-library/blob/master/docs/component-development-guidelines.md) to have examples and docs on using theme provider, theme variables inside of components and theme overrides.

### LINKED ISSUE #205: Docs: Pixels vs. ems in DefaultTheme
Document Pixels vs. Ems:

- Fonts: Use rems and ems
- Everything else: Px

### PR DESCRIPTION
**Pull request by aldeed**
Resolves #118 
Resolves #205
Impact: **minor**  
Type: **docs**

Updates the component style conventions docs.

### CODE REVIEW

**Comment by machikoyasuda** [comment]
🚀 Preview deployed

Built with commit 7cf10fe2796959b2093375c062914a95b64fda8b

https://deploy-preview-310--stoic-hodgkin-c0179e.netlify.com

**Comment by aldeed** [comment]
Listed everyone as reviewers. I've updated these docs to better explain what the theme guidelines were always supposed to be. We've strayed from this quite a bit in some places, but I'll update everything to do it correctly after we all agree on this.

One open question is whether we want to introduce one layer of subobjects on the theme context. So rather than:

```js
{
  rui_componentNameBorderLeftWidth
  rui_componentNameBorderRightWidth
}
```

It would be:

```js
{
  rui_ComponentName: {
    BorderLeftWidth
    BorderRightWidth
  }
}
```

Brent asked for this and I agree it might be better.

File: docs/styling-conventions.md [hunk 1]
```diff
@@ -1,14 +1,156 @@
...
+- border radius
+- font family
+- font size
+- font weight
+
+> ALWAYS use the most specific CSS rule there is. For example, you should never have just a single "Padding" variable, but rather "PaddingLeft", "PaddingRight", "PaddingTop", and "PaddingBottom". Similarly, never use just "Border" or "BorderLeft", but rather the full specific rule such as "BorderLeftColor" or "BorderTopLeftRadius".
+
+In general, flexbox and other layout styles should be baked into the component and not controlled by theme. A component can have props that tell it to use one of several layouts if necessary, or better yet, we can create multiple similar components. Theming should be reserved for borders, colors, padding, and typography.
+
+Height and width should come from theme if the component would still be functional with differing height and width. If the ratio of these two is important to the look or function of the component, then include just a single value in the theme and calculate the other value from that. If the ratio is 1:1, then use a single variable ending in "Size".
```
Comment (willopez) [inline docs/styling-conventions.md]: Not clear on why the height and width of a component should be defined in the theme? can you elaborate on this.

Comment (aldeed) [reply docs/styling-conventions.md]: Not of the component but of something in the component. Two examples we have now are the checkbox height/width and the size of icons or images that appear in the component. I will clarify.

File: docs/styling-conventions.md [hunk 2]
```diff
@@ -136,3 +142,15 @@ Every component that renders text must explicitly set the following typography s
...
+## Margins
+
+- All components must have zero margin on their outermost DOM element.
+- For spacing within the outermost DOM element, prefer "padding" over "margin".
+- If it is absolutely necessary to set the margin on an element within a component, create a theme variable specific to that, but use a more specific word such as "Spacing" instead of "Margin" if possible.
+
+## Units (px | em | rem)
+
+- Almost always use pixel values in the default theme, except...
+- Use `em` only where you specifically need the component style values to expand or contract with the current font size. Explain in the component markdown documentation exactly why you feel `em` is appropriate.
```
[the remark this answers is not in the document]
Comment (aldeed) [reply docs/styling-conventions.md]: Changed it to just "using"

File: docs/styling-conventions.md [hunk 3]
```diff
@@ -1,14 +1,156 @@
...
+The following styles should almost always be put into the theme file as variables, rather than hard coded in the component file:
+- padding
+- colors
+- border width
+- border radius
+- font family
+- font size
+- font weight
+
+> ALWAYS use the most specific CSS rule there is. For example, you should never have just a single "Padding" variable, but rather "PaddingLeft", "PaddingRight", "PaddingTop", and "PaddingBottom". Similarly, never use just "Border" or "BorderLeft", but rather the full specific rule such as "BorderLeftColor" or "BorderTopLeftRadius".
```
Comment (machikoyasuda) [inline docs/styling-conventions.md]: Does this mean... if we're adding a default `10px` padding to all sides, we should declare:
```
PaddingLeft: 10px,
PaddingRight: 10px,
PaddingTop: 10px,
PaddingBottom: 10px
```

Right?

**Comment by aldeed** [comment]
@nnnnat I thought about the displayName thing, too, but I'm not sure how it would really be possible given that we're actually using `applyTheme` on DOM elements that are **_within_** that component, and usually not on the component itself.

`applyTheme` would work with object groupers as currently written if we simply add a period after the component name everywhere (because it uses lodash get).

I'm currently thinking that I'll merge this, PR to update all the styles to follow these conventions, and then do a separate PR just for splitting them into component groups and update these docs at that time.

**Pull request merged**